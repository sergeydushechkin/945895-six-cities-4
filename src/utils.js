const getRatingWidth = (rating) => {
  return Math.round(rating) * 20;
};

export {getRatingWidth};
