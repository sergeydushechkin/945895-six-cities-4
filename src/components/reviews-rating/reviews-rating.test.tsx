import * as React from "react";
import * as renderer from "react-test-renderer";

import {noop} from "../../utils";
import ReviewsRating from "./reviews-rating";

it(`Render ReviewsRating`, () => {
  const tree = renderer
    .create(
        <ReviewsRating
          onRatingChange={noop}
          isFormDisabled={false}
          rating={`5`}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
