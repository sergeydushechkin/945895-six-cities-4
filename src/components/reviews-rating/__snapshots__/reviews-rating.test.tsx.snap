// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ReviewsRating 1`] = `
<div
  className="reviews__rating-form form__rating"
>
  <input
    checked={true}
    className="form__rating-input visually-hidden"
    disabled={false}
    id="5-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={5}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    disabled={false}
    id="4-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={4}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="4-stars"
    title="good"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    disabled={false}
    id="3-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={3}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="3-stars"
    title="not bad"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    disabled={false}
    id="2-stars"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={2}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="2-stars"
    title="badly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    disabled={false}
    id="1-star"
    name="rating"
    onChange={[Function]}
    type="radio"
    value={1}
  />
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-star"
    title="terribly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>
</div>
`;
